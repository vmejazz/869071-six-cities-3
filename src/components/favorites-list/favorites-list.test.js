import React from "react";
import renderer from "react-test-renderer";
import FavoritesList from "./favorites-list.jsx";
import customHistory from "../../history.js";
import {Router} from "react-router-dom";
jest.mock(`../apartment-list/apartment-list.jsx`, () => `ApartmentList`);

const offersFavorite = [
  {
    id: 12,
    title: `House in countryside`,
    price: 405,
    srcImg: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg`,
    srcGallery: [
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`
    ],
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    premium: true,
    type: `house`,
    rate: 3.8,
    bedrooms: 3,
    maxGuests: 8,
    apartmentStuff: [
      `Breakfast`,
      `Laptop friendly workspace`,
      `Washer`
    ],
    ownerInfo: {
      "name": `Angelina`,
      'super': true,
      "srcAvatar": `img/avatar-angelina.jpg`,
      "id": 25
    },
    position: [
      51.211402,
      6.756314000000001
    ],
    city: `Dusseldorf`,
    favorite: true
  },
  {
    id: 4,
    title: `Canal View Prinsengracht`,
    price: 185,
    srcImg: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg`,
    srcGallery: [
    ],
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    premium: false,
    type: `hotel`,
    rate: 2.9,
    bedrooms: 3,
    maxGuests: 8,
    apartmentStuff: [
      `Laptop friendly workspace`,
      `Breakfast`
    ],
    ownerInfo: {
      "name": `Angelina`,
      'super': true,
      "srcAvatar": `img/avatar-angelina.jpg`,
      "id": 25
    },
    position: [
      51.216402,
      6.758314
    ],
    city: `Dusseldorf`,
    favorite: true
  },
];

const cityesFavorite = [
  `Dusseldorf`,
  `Amsterdam`,
  `Cologne`
];


it(`<FavoritesList /> should render favorites list `, () => {
  const tree = renderer
      .create(
          <Router history={customHistory}>
            <FavoritesList
              offersFavorite={offersFavorite}
              cityesFavorite={cityesFavorite}
            />
          </Router>
      )
      .toJSON();

  expect(tree).toMatchSnapshot();
});
