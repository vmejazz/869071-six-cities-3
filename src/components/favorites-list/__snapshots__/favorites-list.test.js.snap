// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavoritesList /> should render favorites list  1`] = `
<ul
  className="favorites__list"
>
  <li
    className="favorites__locations-items"
  >
    <div
      className="favorites__locations locations locations--current"
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link"
          href="/"
          onClick={[Function]}
        >
          <span>
            Dusseldorf
          </span>
        </a>
      </div>
    </div>
    <ApartmentList
      isFavoriteList={true}
      offersShow={
        Array [
          Object {
            "apartmentDetails": Array [
              "Breakfast",
              "Laptop friendly workspace",
              "Washer",
            ],
            "bedrooms": 3,
            "city": "Dusseldorf",
            "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
            "favorite": true,
            "id": 12,
            "maxGuests": 8,
            "ownerInfo": Object {
              "id": 25,
              "name": "Angelina",
              "srcAvatar": "img/avatar-angelina.jpg",
              "super": true,
            },
            "positions": Array [
              51.211402,
              6.756314000000001,
            ],
            "premium": true,
            "price": 405,
            "rate": 3.8,
            "imageURLs": Array [
              "https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg",
            ],
            "srcImg": "https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg",
            "title": "House in countryside",
            "type": "house",
          },
          Object {
            "apartmentDetails": Array [
              "Laptop friendly workspace",
              "Breakfast",
            ],
            "bedrooms": 3,
            "city": "Dusseldorf",
            "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
            "favorite": true,
            "id": 4,
            "maxGuests": 8,
            "ownerInfo": Object {
              "id": 25,
              "name": "Angelina",
              "srcAvatar": "img/avatar-angelina.jpg",
              "super": true,
            },
            "positions": Array [
              51.216402,
              6.758314,
            ],
            "premium": false,
            "price": 185,
            "rate": 2.9,
            "imageURLs": Array [],
            "srcImg": "https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg",
            "title": "Canal View Prinsengracht",
            "type": "hotel",
          },
        ]
      }
      onCardHover={[Function]}
    />
  </li>
  <li
    className="favorites__locations-items"
  >
    <div
      className="favorites__locations locations locations--current"
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link"
          href="/"
          onClick={[Function]}
        >
          <span>
            Amsterdam
          </span>
        </a>
      </div>
    </div>
    <ApartmentList
      isFavoriteList={true}
      offersShow={Array []}
      onCardHover={[Function]}
    />
  </li>
  <li
    className="favorites__locations-items"
  >
    <div
      className="favorites__locations locations locations--current"
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link"
          href="/"
          onClick={[Function]}
        >
          <span>
            Cologne
          </span>
        </a>
      </div>
    </div>
    <ApartmentList
      isFavoriteList={true}
      offersShow={Array []}
      onCardHover={[Function]}
    />
  </li>
</ul>
`;
